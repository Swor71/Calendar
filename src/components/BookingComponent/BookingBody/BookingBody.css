.booking-body__wrapper {
  padding-top: 20px;
}

.booking-body__header {
  font-weight: 600;
  font-size: .9em;
  margin-bottom: 5px;
}

.booking-body__arrow {
  font-size: 1.2em;
  padding: 0 20px;
  vertical-align: top;
}

.booking-body__inputs {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border: 2px solid #dee0e4;
  border-radius: 4px;
  padding: 10px;
}

[class*="booking-body__inputs--check-"] {
  border: transparent;
  width: 50%;
  border-radius: 4px;
  padding: 5px;
  outline: none;
  font-size: 1.2em;
}

[class*="booking-body__inputs--check-"]:focus {
  background-color: paleturquoise;
}
