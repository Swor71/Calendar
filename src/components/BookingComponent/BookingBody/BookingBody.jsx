import React, { PureComponent } from 'react';
import CalendarComponent from '../../CalendarComponent/CalendarComponent';
import PropTypes from 'prop-types';
import { calculateAmountOfDays, calculateDiffInMS } from '../../../utils/dateFormats';

import './BookingBody.css';

class BookingBody extends PureComponent {
  static propTypes = {
    availableDates: PropTypes.arrayOf(PropTypes.object),
    calendarData: PropTypes.objectOf(PropTypes.any),
    onHandleSelectDate: PropTypes.func,
    onHandleMonthChange: PropTypes.func,
    onHandleShowCalendar: PropTypes.func,
    price: PropTypes.number,
  }

  render() {
    const { calendarData, calendarData: { checkInDate, checkOutDate, showCalendar }, availableDates, onHandleSelectDate, onHandleMonthChange, onHandleShowCalendar, price } = this.props;

    let lenghtOfStay = 0;

    if (checkInDate && checkOutDate) {
      const ms = calculateDiffInMS(checkInDate, checkOutDate);
      lenghtOfStay = calculateAmountOfDays(ms);
    }

    return (
      <div className="booking-body__wrapper">
        <div className="booking-body__header">Dates</div>
        <div className="booking-body__inputs">
          <input
            onClick={() => onHandleShowCalendar('checkInDate')}
            readOnly
            type="text"
            placeholder="Check In"
            className="booking-body__inputs--check-in"
            value={checkInDate}
          />
          <span className="booking-body__arrow">⟶</span>
          <input
            onClick={() => onHandleShowCalendar('checkOutDate')}
            readOnly
            type="text"
            placeholder="Check Out"
            className="booking-body__inputs--check-out"
            value={checkOutDate}
          />
          {showCalendar
            ? <CalendarComponent
              calendarData={calendarData}
              availableDates={availableDates}
              onHandleSelectDate={onHandleSelectDate}
              onHandleMonthChange={onHandleMonthChange}
            />
            : null}
        </div>
        <div className="booking-body__output-wrapper">
          <span className="booking-body__output--header">Total price</span>
          <div className="booking-body__output--total-price">
            <span className={checkInDate && checkOutDate ? 'price-calculated' : ''}>
              {checkInDate && checkOutDate && lenghtOfStay !== 0
                ? `Total price for ${lenghtOfStay} night${lenghtOfStay > 1 ? 's' : ''} is ${lenghtOfStay * price} zł`
                : 'Please select both dates'}</span>
          </div>
        </div>
        <button className="booking-body__confirmation-button">confirm your booking</button>
      </div>
    );
  }
}

export default BookingBody;
