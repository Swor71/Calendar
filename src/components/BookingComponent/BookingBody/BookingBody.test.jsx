import React from 'react';
import BookingBody from './BookingBody';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('BookingBody', () => {
  it('renders the component', () => {
    const component = shallow(<BookingBody />);
    expect(component.length).toEqual(1);
  });

  it('renders correctly', () => {
    const component = renderer
      .create(<BookingBody />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
