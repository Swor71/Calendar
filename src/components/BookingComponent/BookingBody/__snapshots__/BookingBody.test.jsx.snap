// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingBody renders correctly 1`] = `
<div
  className="booking-body__wrapper"
>
  <div
    className="booking-body__header"
  >
    Dates
  </div>
  <div
    className="booking-body__inputs"
  >
    <input
      className="booking-body__inputs--check-in"
      onClick={[Function]}
      placeholder="Check In"
      readOnly={true}
      type="text"
      value=""
    />
    <span
      className="booking-body__arrow"
    >
      ‚ü∂
    </span>
    <input
      className="booking-body__inputs--check-out"
      onClick={[Function]}
      placeholder="Check Out"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="booking-body__output-wrapper"
  >
    <span
      className="booking-body__output--header"
    >
      Total price
    </span>
    <div
      className="booking-body__output--total-price"
    >
      <span
        className=""
      >
        Please select both dates
      </span>
    </div>
  </div>
  <button
    className="booking-body__confirmation-button"
  >
    confirm your booking
  </button>
</div>
`;
